pkgs/development/libraries/qt-6/modules/qtbase/default.nix

```diff
@@ -163,10 +164,13 @@ stdenv.mkDerivation rec {
     ++ lib.optionals libGLSupported [
       libGL
     ]
-    ++ lib.optionals stdenv.hostPlatform.isMinGW [
+    ++ lib.optionals vulkanSupport [
       vulkan-headers
       vulkan-loader
     ]
+    ++ lib.optionals (vulkanSupport && stdenv.hostPlatform.isDarwin) [
+      moltenvk
+    ]
```

Comment:
`moltenvk` shouldn’t be required on Darwin; `vulkan-loader` hardcodes it.

`vulkan-loader` is apparently available on `platforms.unix ++ platforms.windows`, i.e. everything we can build Qt for. I think we can just make these unconditional and drop the flag, but let’s see what @K900 thinks.

(I’m not sure why OpenGL is conditional on Linux either; I guess it’s downstream of when `libGL` used to be bad to use on Darwin?)

Reply:
I didn't think MoltenVK should have been necessary either, but it fails to build the `qtbase` needed by `qttools` complaining about missing MoltenVK headers (in `src/plugins/platforms/cocoa/qcocoawindow.h`) if it's not passed as input.

Reply:
It does in fact pull internal MoltenVK headers for some reason: https://github.com/qt/qtbase/blob/dev/src/plugins/platforms/cocoa/qcocoawindow.h#L19

libGL shouldn't be conditional, I think.

Reply:
So would you say I should just go ahead and make both unconditional in this PR?

Reply:
I think we should be fine, yes. MoltenVK needs to stay conditional on Darwin though.